fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/filter/LoanCOBApiFilter.java

```diff
@@ -206,6 +211,9 @@ private Long getLoanId(boolean isGlim, String pathInfo) {
             if (isExternal(pathInfo)) {
                 String externalId = id;
                 return loanRepository.findIdByExternalId(new ExternalId(externalId));
+            } else if (isRescheduleLoans(pathInfo)) {
+                return loanRescheduleRequestRepository.findById(Long.valueOf(id)).map(LoanRescheduleRequest::getLoan)
```

Comment:
I think we should avoid to fetch the full loan here...

Reply:
Yep, let's only fetch the ID of the loan. We don't even need to fetch the loan reschedule request either.